### PR TITLE
Update all libraries and tests

### PR DESCRIPTION
**Pull request by justin808**
* bootstrap-loader requires breaking changes of removal of Input
* Tests not passing

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/334)

### CODE REVIEW

**Comment by justin808** [comment]
Anybody want to take this one? Comment in this thread.

**Comment by justin808** [comment]
Actually, updating bootstrap-loader might fix the rspec failures, as there are many deprecation warnings.

**Comment by justin808** [comment]
* https://github.com/react-bootstrap/react-bootstrap/pull/1765
* http://react-bootstrap.github.io/components.html#forms-controls

I'm going to first try: https://gist.github.com/Bazze/81cf778d8843948977421bea5c8cdfee

**Comment by Judahmeek** [comment]
I'd like to help with this. I noticed that Travis is failing because selenium_firefox is not headless. A google search led me to https://docs.travis-ci.com/user/gui-and-headless-browsers/ and https://saucelabs.com/open-source. Do you think saucelabs would be a viable option for us or would you rather I focus on webkit and poltergeist?

**Comment by justin808** [comment]
> I'd like to help with this. I noticed that Travis is failing because selenium_firefox is not headless. A google search led me to https://docs.travis-ci.com/user/gui-and-headless-browsers/ and https://saucelabs.com/open-source. Do you think saucelabs would be a viable option for us or would you rather I focus on webkit and poltergeist?

* Headless should work.
* I'm open to trying https://saucelabs.com/open-source since they have a free open source option.

**Comment by justin808** [comment]
@Judahmeek Any progress on this one?

**Comment by justin808** [comment]
@Judahmeek I incorporated your changes, plus many more.

File: .travis.yml [hunk 1]
```diff
@@ -4,25 +4,39 @@ rvm:
...
     packages:
       - g++-4.9
-      - sqlite3
+services:
+  - postgresql
 env:
   global:
     - RAILS_ENV=test
     - CXX=g++-4.9
+    - SAUCE_LABS_URL=http://judahmeek:1ed6d81c-c3e0-4fd0-9012-1d2274c4b261@ondemand.saucelabs.com:80/wd/hub
```
Comment (justin808) [inline .travis.yml]: remove sauce

File: .travis.yml [hunk 2]
```diff
@@ -4,25 +4,39 @@ rvm:
...
+  - mkdir $PWD/travis-phantomjs
+  - curl -sSL https://github.com/Medium/phantomjs/releases/download/v2.1.1/phantomjs-2.1.1-linux-x86_64.tar.bz2 -o $PWD/travis-phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2
+  - tar -xvf $PWD/travis-phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2 -C $PWD/travis-phantomjs
+  - export PATH=$PWD/travis-phantomjs/phantomjs-2.1.1-linux-x86_64/bin:$PATH
 install:
   - bundle install
-  - nvm install 6.3
-  - nvm use 6.3
+  - nvm install 6.7
+  - nvm use 6.7
```
Comment (justin808) [inline .travis.yml]: maybe update to 6.9.1

File: Gemfile [hunk 3]
```diff
@@ -104,7 +100,7 @@ group :test  do
   gem "generator_spec"
   gem "launchy"
   gem "poltergeist"
-  gem "rspec-rails", "3.5.0.beta3"
+  gem "rspec-rails", "~> 3"
   gem "rspec-retry"
-  gem "selenium-webdriver", require: !["poltergeist", "poltergeist_errors_ok", "webkit"].include?(ENV["DRIVER"])
+  gem "selenium-webdriver", "<3.0.0", require: !["poltergeist", "poltergeist_errors_ok", "webkit"].include?(ENV["DRIVER"])
```
Comment (justin808) [inline Gemfile]: @Judahmeek why less than 3.0.0

File: spec/support/driver_registration.rb [hunk 4]
```diff
@@ -1,19 +1,33 @@
...
+  #   Capybara.register_driver :selenium_chrome do |app|
+  #     Capybara::Selenium::Driver.new(app, browser: :chrome)
+  #   end
+  #   Capybara::Screenshot.register_driver(:selenium_chrome) do |js_driver, path|
+  #     js_driver.browser.save_screenshot(path)
+  #   end
+  #   @selenium_chrome_registered = true
+  # end
+
+  def self.driver_caps(app)
```
Comment (justin808) [inline spec/support/driver_registration.rb]: remove sauce

so sauce can be a small PR so people can reference it

nobody wants to reference a 6K line PR

File: Gemfile (hunk 3 above)
Comment (Judahmeek) [inline Gemfile]: That was for sauce compatibility. They don't fully support Selenium 3 yet. Their docs certainly don't.

**Comment by justin808** [comment]
Sometimes passing!

**Comment by justin808** [comment]
@Judahmeek we can do the sauce thing as a separate PR

**Pull request merged**